Utils/azuretools-core/src/com/microsoft/azuretools/ui/embeddedbrowser/JxBrowserUtil.java

```diff
@@ -22,42 +22,303 @@
+        File targetFile = filePathName.toFile();
+        if (!targetFile.exists() || !targetFile.isFile()) {
+            log.warning(filePathName + " doesn't exist");
+            return false;
+        }
+
+        File digestFile = digestFilePathName.toFile();
+        if (!digestFile.exists() || !digestFile.isFile()) {
+            log.warning(digestFilePathName + " doesn't exist");
+            return false;
+        }
+
+        FileInputStream targetFileStream = null;
+        String digestFileContent = "content";
+        String digest = "digest";
+        boolean result = true;
+        try {
+            targetFileStream = new FileInputStream(targetFile);
+            digest = DigestUtils.md5Hex(targetFileStream).toUpperCase();
+            digestFileContent = FileUtils.readFileToString(digestFile, "UTF-8");
+        } catch (IOException e) {
+            log.warning(String.format("Fail to read file %s or digest %s", filePathName, digestFilePathName));
+            result = false;
+        } finally {
+            IOUtils.closeQuietly(targetFileStream);
+        }
+
+        return result == true && digestFileContent.equals(digest);
+    }
+
+    /**
+     * Download file from Azure and save to target path specified. Return true if download succeeds
+     * @param filePathName
+     * @return
+     */
+    private static Boolean downloadFromAzure(@NotNull Path filePathName) {
+        Boolean result = true;
+        try {
+            URI cloudBlobUri = new URI(AZURE_BLOB_URI);
+            CloudBlobClient serviceClient = new CloudBlobClient(cloudBlobUri);
+            CloudBlobContainer container = serviceClient.getContainerReference("libcontainer1");
+            CloudBlockBlob blob = container.getBlockBlobReference(filePathName.getFileName().toString());
+            File downloadingFile = filePathName.toFile();
+            blob.downloadToFile(downloadingFile.getAbsolutePath());
+        } catch (Exception e) {
+            result = false;
+            log.warning("Fail to download file from Azure: " + e.getMessage());
+        }
+
+        return result;
+    }
+
+    /**
+     * Download file and check digest with retry. Return false if all retry fails
+     * @param threadPool
+     * @param filePathNames
+     * @param currentRetry
+     * @param waitTime
+     * @return
+     */
+    private static boolean downloadAndCheckDigest(@NotNull ExecutorService threadPool, @NotNull List<Path> filePathNames, int currentRetry, int waitTime) {
+        List<Path> digestPathNames = new ArrayList<>();
+        for (int i = 0; i < filePathNames.size(); i++) {
+            digestPathNames.add(Paths.get(FilenameUtils.removeExtension(filePathNames.get(i).toString()) + ".md5"));
+        }
+
+        boolean result = true;
+        try {
+            int i = filePathNames.size() - 1;
+            for (; i >= 0; i--) {
+                if (checkFileDigest(filePathNames.get(i), digestPathNames.get(i)) == true)  {
```

Comment:
>== true [](start = 82, length = 7)

remove it?

Reply:
Fixed

---
In reply to: [156552795](https://github.com/Microsoft/azure-tools-for-java/pull/1296#discussion_r156552795) [](ancestors = 156552795)